src/ray/rpc/gcs_server/gcs_rpc_server.h

```diff
@@ -382,7 +381,11 @@ class NodeInfoGrpcService : public GrpcService {
       const std::unique_ptr<grpc::ServerCompletionQueue> &cq,
       std::vector<std::unique_ptr<ServerCallFactory>> *server_call_factories,
       const ClusterID &cluster_id) override {
-    NODE_INFO_SERVICE_RPC_HANDLER(GetClusterId);
+    RPC_SERVICE_HANDLER_CUSTOM_AUTH(
+        NodeInfoGcsService,
+        GetClusterId,
+        RayConfig::instance().gcs_max_active_rpcs_per_handler(),
+        AuthType::LAZY_AUTH);
```

Comment:
This should be NO_AUTH?

Reply:
I changed it, but I'm not sure there's a clear answer to whether we want to allow raylets that originally belonged to a different cluster to rejoin a new cluster.

Reply:
Actually, let's keep it this way to enforce that raylets can only have one cluster ID in their life cycle?